### ISSUE TITLE
HTTPBearer security scheme is returning 403 instead or 401

### ISSUE THREAD

**Issue by Kludex** [fastapi/fastapi#10177]
### Discussed in https://github.com/tiangolo/fastapi/discussions/9130

<div type='discussions-op-text'>

<sup>Originally posted by **aaaaahaaaaa** September  8, 2020</sup>
HTTPBearer security scheme enabled as a dependency is returning a `403` when a request is unauthenticated because of a missing or a malformed `authorization` header. In those scenarios, a `401` should be returned instead.</div>

## Related PRs
- https://github.com/tiangolo/fastapi/pull/2120
- https://github.com/tiangolo/fastapi/pull/5310
- https://github.com/tiangolo/fastapi/pull/5332
- https://github.com/tiangolo/fastapi/pull/9312

**Comment by ezeparziale** [comment]
I use the `auto_error=False` and control the correct http error code manually.

**Comment by Kludex** [comment]
Can you show me the diff you are talking about?

**Comment by ordinary-jamie** [comment]
[`fastapi/security/http.py` L#76](https://github.com/tiangolo/fastapi/blob/master/fastapi/security/http.py#L76-L82)

**Comment by Kludex** [comment]
You want 400 instead of 403? That's why I asked the diff, not the lines in the code source. 👀

**Comment by ordinary-jamie** [comment]
Sorry, my misunderstanding. At any rate, I don't think it is necessary after reading section 3 again I realised they offered an example for this scenario.

**tldr; 401 seems like the appropriate error code 👍** 

```
 If the request lacks any authentication information (e.g., the client
   was unaware that authentication is necessary or attempted using an
   unsupported authentication method), the resource server SHOULD NOT
   include an error code or other error information.

   For example:

     HTTP/1.1 401 Unauthorized
     WWW-Authenticate: Bearer realm="example"
```

**Comment by bootrecords** [comment]
This not only affects the `HTTPBearer`, but a lot of other classes in the `security` scope also return 403 in `"Not authenticated"` cases where they should return 401 instead, such as the `OAuth2` class (though strangely enough not the subclasses `OAuth2PasswordBearer` and `OAuth2AuthorizationCodeBearer`), the `HTTPBase` class and its sublasses (except for `HTTPBasic` for some reason), all classes in `api_key.py` and `OpenIdConnect`. Would probably be good if that could be straightened out for all occurrences there.

**Comment by eastack** [comment]
Will there be any progress in the future? Will this error be fixed? I just hope that when the client receives a 403 response, it prompts the user that they do not have access to the corresponding resource, and when a 401 response is received, it redirects to the login page. It seems that this cannot be achieved with the current default implementation of FastAPI.

**Comment by YuriiMotov** [comment]
For those who are interested in getting this issue resolved, please review #13786

**Comment by tiangolo** [comment]
This was handled in https://github.com/fastapi/fastapi/pull/13786, available in FastAPI 0.122.0 🎉

**Comment by asmaier** [comment]
This change broke some of our tests. As a workaround limit the version to `fastapi[all]<0.122.0"` or fix your tests to expect now 401 instead of 403.

**Comment by YuriiMotov** [comment]
> This change broke some of our tests. As a workaround limit the version to `fastapi[all]<0.122.0"` or fix your tests to expect now 401 instead of 403.

Consider also https://fastapi.tiangolo.com/how-to/authentication-error-status-code/
